builder/dockerfile/copy.go

```diff
@@ -82,14 +83,20 @@ func (o *copier) createCopyInstruction(args []string, cmdName string) (copyInstr
 	inst := copyInstruction{cmdName: cmdName}
 	last := len(args) - 1
 
-	// Work in daemon-specific filepath semantics
-	inst.dest = filepath.FromSlash(args[last])
+	// TODO: @gupta-ak. The platform needs to be here to support LCOW + WCOW. Or in the copier constructor.
+	separator := "/"
+	if !system.LCOWSupported() {
+		// Work in daemon-specific filepath semantics
+		inst.dest = filepath.FromSlash(args[last])
+		separator = string(os.PathSeparator)
+	}
```

Comment:
For the most part the path details seem to be well abstracted, but they leak out in this function.

Instead of this maybe we could have a `system.HasDirectorySuffix()` and `system.FromSlash()` or something like that?

`system` is already a large package, so maybe that's not the right place for it.